go/host/host.go

```diff
@@ -847,29 +847,27 @@ func (h *host) handleBatches(encodedBatches *common.EncodedBatches) error {
 		return nil
 	}
 
-	// We store the batches.
-	err = h.batchManager.StoreBatches(batches)
-	if err != nil {
-		if !errors.Is(err, batchmanager.ErrBatchesMissing) {
-			return fmt.Errorf("could not store batches. Cause: %w", err)
-		}
-
-		// We have encountered missing batches. We abort the storage operation and request the missing batches.
-		batchRequest, err := h.batchManager.CreateBatchRequest(h.config.P2PPublicAddress)
+	// We store the batches and submit them to the enclave.
+	for _, batch := range batches {
+		err = h.batchManager.StoreBatch(batch)
 		if err != nil {
-			return fmt.Errorf("could not create batch request. Cause: %w", err)
-		}
-		if err = h.p2p.RequestBatches(batchRequest); err != nil {
-			return fmt.Errorf("could not request historical batches. Cause: %w", err)
-		}
+			if !errors.Is(err, batchmanager.ErrBatchesMissing) {
+				return fmt.Errorf("could not store batches. Cause: %w", err)
+			}
 
-		// If we requested any batches, we return early and wait for the missing batches to arrive.
-		return nil
-	}
+			// We have encountered missing batches. We abort the storage operation and request the missing batches.
+			batchRequest, err := h.batchManager.CreateBatchRequest(h.config.P2PPublicAddress)
+			if err != nil {
+				return fmt.Errorf("could not create batch request. Cause: %w", err)
+			}
+			if err = h.p2p.RequestBatches(batchRequest); err != nil {
+				return fmt.Errorf("could not request historical batches. Cause: %w", err)
+			}
+			return nil
+		}
 
-	// TODO - #718 - We should probably submit each batch after storing it, and not submitting each one only if *every*
-	//  batch was stored correctly.
-	for _, batch := range batches {
+		// TODO - #718 - What happens if this op fails? We'll think we have the batch, so we'll never catch it up and
```

Comment:
So shouldn't we store the batch only *after* successful submission to Enclave?

Reply:
Yeah, will reorder.